src/daft-functions/src/list/contains.rs

```diff
@@ -0,0 +1,65 @@
+use common_error::{DaftError, DaftResult};
+use daft_core::{
+    prelude::{DataType, Field, Schema},
+    series::Series,
+};
+use daft_dsl::{
+    functions::{ScalarFunction, ScalarUDF},
+    ExprRef,
+};
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq, Hash)]
+pub struct ListContains;
+
+#[typetag::serde]
+impl ScalarUDF for ListContains {
+    fn as_any(&self) -> &dyn std::any::Any {
+        self
+    }
+
+    fn name(&self) -> &'static str {
+        "list_contains"
+    }
+
+    fn to_field(&self, inputs: &[ExprRef], schema: &Schema) -> DaftResult<Field> {
+        let [input, contains] = inputs else {
+            return Err(DaftError::SchemaMismatch(format!(
+                "Expected 2 input args, got {}",
+                inputs.len()
+            )));
+        };
+        let input = input.to_field(schema)?;
+        let (DataType::List(internal_dt) | DataType::FixedSizeList(internal_dt, _)) = input.dtype
+        else {
+            return Err(DaftError::TypeError(format!(
+                "Expected input to be a list type, received: {}",
+                input.dtype,
+            )));
+        };
+        let contains = contains.to_field(schema)?;
+        if internal_dt.as_ref() != &contains.dtype {
```

Comment:
We should be able to expand the range of comparable dtypes for list contains. Right now for example you wont be able to list contains a list of int32 with an int64 for example, or a datetime with a utf8. 

You can use `comparison_op` in `infer_datatype.rs`, pass in the two dtypes and it will tell you whether it's comparable or not.